api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1178,19 +1178,18 @@ private void updateRegionCache(String bucketName)
       XmlPullParser xpp = xmlPullParserFactory.newPullParser();
       String location = null;
 
-      xpp.setInput(response.body().charStream());
-      while (xpp.getEventType() != XmlPullParser.END_DOCUMENT) {
-        if (xpp.getEventType() ==  XmlPullParser.START_TAG && "LocationConstraint".equals(xpp.getName())) {
+      try (ResponseBody body = response.body()) {
+        xpp.setInput(body.charStream());
+        while (xpp.getEventType() != XmlPullParser.END_DOCUMENT) {
+          if (xpp.getEventType() == XmlPullParser.START_TAG && "LocationConstraint".equals(xpp.getName())) {
+            xpp.next();
+            location = getText(xpp);
+            break;
+          }
           xpp.next();
-          location = getText(xpp, location);
-          break;
         }
-        xpp.next();
       }
 
-      // close response body.
-      response.body().close();
```

Comment:
Is the body being closed automatically here?

Reply:
Yes, it's in the enclosed try-using block. Please see https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html